src/test/test_utils.rs

```diff
@@ -163,3 +163,58 @@ pub fn get_errno_message(errno: i32) -> String {
     }
     cstr.to_string_lossy().into_owned()
 }
+
+/// Calls check_system_call(), but automatically passes the current line number.
+#[macro_export]
+macro_rules! check_system_call {
```

Comment:
Not sure I understand the motivation for passing the line number to `check_system_call` and wrapping it in a macro to do so. If the caller were to do `check_system_call(f, expected_errnos).unwrap()`, wouldn't that achieve the same?

Reply:
All of the socket tests have the `summarize` option which should run all of the tests, even if some fail. So you can't unwrap results, and instead need to propagate them which means you lose the location of the error. The `--summarize` option is useful for debugging, but maybe it's not worth the added code complexity.

Reply:
Ahhh ok.

I'm trying to think how we can avoid writing macros for every such helper function. Given that this is the first time it's come up, maybe it's fine to proceed as-is.

Another option might be have a utility function that takes the list of tests, config, and runs them (this might be a good idea anyway), and just have it catch panics in individual tests and stuff them into an error for the summary. This would let us use built in assert helpers, unwrap, etc.

Reply:
I thought about this a long time ago, but my reason at the time was that you probably also want to suppress the panic message (including the backtrace if the environment variable is set), so you need to also set the global panic hook which can cause issues in rust's standard test runner which uses multi-threading. But since we aren't using rust's test runner and are not using multi-threading, this may be fine to do.